contentprovider.go

```diff
@@ -664,23 +683,36 @@ func (p *contentProvider) candidateMatchScore(mts matchTreeScorer, ms []*candida
 			}
 		}
 
-		if score.score > maxScore.score {
-			maxScore.score = score.score
-			maxScore.what = score.what
+		scoreIdx := bits.TrailingZeros64(m.queryAtoms)
```

Comment:
Can you explain this part where we calculate (and clamp) `scoreIdx`?

Reply:
treat this as me just trying to get something to work. This is based of my other PR which introduces queryAtoms which is a bitset (not a counter). So we do this clamping since we can't support more than 64 atoms.